rfcs/0000-limitador-multithread-inmemory.md

```diff
@@ -0,0 +1,241 @@
+# RFC Template
+
+- Feature Name: `limitador_multithread_inmemory`
+- Start Date: 2023-11-02
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+
+# Summary
+[summary]: #summary
+
+Enable Limitador service to process requests in parallel when configured to use In Memory storage.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Limitador service is single threaded, regardless of its chosen storage. This means that it can only process one request at a time. 
```

Comment:
This is actually not true, is it? The gRPC service (tonic) is built on top of Tokio, an event-driven, non-blocking I/O platform. Or when the storage is memory, there is no I/O involved, hence only one request at a time can be processed?

Reply:
See [this issues](https://github.com/Kuadrant/limitador/issues/14), other than something having changed in newer version, this was still true like 10 months ago.